filebeat/input/input.go

```diff
@@ -164,3 +167,11 @@ func (p *Runner) stop() {
 func (p *Runner) String() string {
 	return fmt.Sprintf("input [type=%s]", p.config.Type)
 }
+
+func (p *Runner) SetStatusReporter(statusReporter status.StatusReporter) {
+	p.statusReporter = statusReporter
+}
+
+func (p *Runner) GetStatusReporter() status.StatusReporter {
+	return p.statusReporter
+}
```

Comment:
Why do we need these 2 functions? We could just make this field public then.

Reply:
@rdner If you have a look at this snippet introduced in https://github.com/elastic/beats/pull/39209 https://github.com/elastic/beats/blob/1a6318d215517feb4c5c880619db8589d926956b/libbeat/management/status/status.go#L52-L55 
All the components supporting status reporter should implement `SetStatusReporter` method  

The way `SetStatusReporter` gets called can be found here https://github.com/elastic/beats/blob/1a6318d215517feb4c5c880619db8589d926956b/libbeat/cfgfile/list.go#L157-L161 in a generic way for all the runners.

Reply:
@rdner ignore above comment, added my mistake.